src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,259 @@
+  },
+
+  onButtonChanged: function (evt) {
+    var button = this.mapping[this.data.hand]['button' + evt.detail.id];
+    var buttonMeshes = this.buttonMeshes;
+    var value;
+    // at the moment, if trigger or grip,
+    // touch events aren't happening (touched is stuck true);
+    // synthesize touch events from very low values
+    if (button === 'trigger' || button === 'grip') {
+      var lastValue = this.previousButtonValues[button];
+      var lastFakeTouch = false;
+      if (lastValue) { lastFakeTouch = (lastValue >= FAKE_TOUCH_THRESHOLD); }
+      var thisValue = evt.detail.state.value;
+      var thisFakeTouch = false;
+      if (thisValue) { thisFakeTouch = (thisValue >= FAKE_TOUCH_THRESHOLD); }
+      this.previousButtonValues[button] = thisValue;
+      if (thisFakeTouch !== lastFakeTouch) {
+        if (thisFakeTouch) {
+          this.onButtonTouchStart(evt);
+        } else {
+          this.onButtonTouchEnd(evt);
+        }
+      }
+    }
+    if (typeof button === 'undefined' || typeof buttonMeshes === 'undefined') { return; }
+    if (button !== 'trigger' || !buttonMeshes || !buttonMeshes.trigger) { return; }
+    value = evt.detail.state.value;
+    buttonMeshes.trigger.rotation.x = -value * (Math.PI / 12);
+  },
+
+  onModelLoaded: function (evt) {
+    var controllerObject3D = evt.detail.model;
+    var buttonMeshes;
+    if (!this.data.model) { return; }
+
+    var leftHand = this.data.hand === 'left';
+    buttonMeshes = this.buttonMeshes = {};
+
+    buttonMeshes.grip = controllerObject3D.getObjectByName(leftHand ? 'grip tooche1 group3' : 'grip tooche group4');
+    buttonMeshes.thumbstick = controllerObject3D.getObjectByName(leftHand ? 'tooche1 group3 control_surface group2 thumb_stick' : 'tooche group4 control_surface group2 thumb_stick');
+    buttonMeshes.trigger = controllerObject3D.getObjectByName(leftHand ? 'tooche1 group3 trigger' : 'tooche group4 trigger');
+    buttonMeshes['X'] = controllerObject3D.getObjectByName('tooche1 group3 control_surface group2 button2');
+    buttonMeshes['A'] = controllerObject3D.getObjectByName('tooche group4 control_surface group2 button2');
+    buttonMeshes['Y'] = controllerObject3D.getObjectByName('tooche1 group3 control_surface group2 button3');
+    buttonMeshes['B'] = controllerObject3D.getObjectByName('tooche group4 control_surface group2 button3');
+    buttonMeshes.surface = controllerObject3D.getObjectByName(leftHand ? 'tooche1 group3 face control_surface group2' : 'tooche group4 face control_surface group2');
+
+    // Offset pivot point
+    controllerObject3D.position.set(0, -0.015, 0.04);
+  },
+
+  onButtonEvent: function (id, evtName) {
+    var buttonName = this.mapping[this.data.hand]['button' + id];
+    var i;
+    if (Array.isArray(buttonName)) {
+      for (i = 0; i < buttonName.length; i++) {
+        this.el.emit(buttonName[i] + evtName);
+      }
+    } else {
+      this.el.emit(buttonName + evtName);
+    }
+    if (Array.isArray(buttonName)) {
```

Comment:
LOL if you look back at the comments, we already have one that does something slightly different per your request.  but I see you would like symmetry with vive-controls, will take a look